moler/cmd/unix/sudo.py

```diff
@@ -80,35 +81,62 @@ def start(self, timeout=None, *args, **kwargs):
             self.timeout_from_embedded_command = False
         return super(Sudo, self).start(timeout=timeout, args=args, kwargs=kwargs)
 
-    def _process_embedded_command(self, line, is_full_line):
+    def data_received(self, data):
+        """
+        Overrides data_received to filter some information to embedded command.
+
+        :param data: List of strings sent by device.
+        :return: None
+        """
+        lines = data.splitlines(True)
+        for line in lines:
+            partial_data = line
+            if self._last_not_full_line is not None:
+                line = "{}{}".format(self._last_not_full_line, line)
+                self._last_not_full_line = None
+            is_full_line = self.has_endline_char(line)
+            if is_full_line:
+                line = self._strip_new_lines_chars(line)
+            else:
+                self._last_not_full_line = line
```

Comment:
move that code chunk into method  `line = update_from_cached_incomplete_line(line)` to let it be reused by `Sudo`  and `CommandTextualGeneric`.